File: src/main/java/com/guidewire/tools/chronos/client/ServerResponse.java
package com.guidewire.tools.chronos.client;

public interface ServerResponse {
  int statusCode();
  int statusCodeRangeStart();
  int statusCodeRangeEnd();
  boolean success();
  boolean validationAllowed();

  int getStatusCode();
  int getStatusCodeRangeStart();
  int getStatusCodeRangeEnd();
  boolean isSuccess();
  boolean isValidationAllowed();
}

File: src/main/scala/com/guidewire/tools/chronos/client/ServerResponse.scala
package com.guidewire.tools.chronos.client

/**
 * @author - Eran Withana (ewithana@comprehend.com) 
 */
trait ServerResponse {
  def statusCode: Int
  def statusCodeRangeStart: Int
  def statusCodeRangeEnd: Int
  def success: Boolean
  def validationAllowed: Boolean

  def getStatusCode: Int
  def getStatusCodeRangeStart: Int
  def getStatusCodeRangeEnd: Int
  def isSuccess: Boolean
  def isValidationAllowed: Boolean
}